pkg/storage/bloom/v1/builder.go

```diff
@@ -275,31 +295,26 @@ func (mb *MergeBuilder) processNextSeries(
 	ch := make(chan *BloomCreation)
 	go mb.populate(nextInStore, preExistingBlooms, chunksToAdd, ch)
 
-	for bloom := range ch {
-		if bloom.Err != nil {
-			return nil, bytesAdded, 0, false, false, errors.Wrap(bloom.Err, "populating bloom")
+	for creation := range ch {
+		if creation.Err != nil {
+			return nil, stats.SourceBytes, 0, false, false, errors.Wrap(creation.Err, "populating bloom")
 		}
-		offset, err := builder.AddBloom(bloom.Bloom)
+		offset, err := builder.AddBloom(creation.Bloom)
```

Comment:
Instead of having the stats inside BloomCreation, may AddBloom return the stats?

Reply:
Not sure what you mean. `AddBloom()` appends/encodes the provided bloom to the `blooms` "file".

Reply:
You're right, I missunderstood what was going on on this loop. The stats come from the channel (through `creation`). Resolving this.